jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/Parser.java

```diff
@@ -324,28 +338,28 @@ private ParsedFrame parsePayload(ByteBuffer buffer)
         {
             if (available < payloadLength)
             {
-                // not enough to complete this frame
-                // Can we auto-fragment
+                // Not enough data to complete this frame, can we auto-fragment?
                 if (configuration.isAutoFragment() && isDataFrame)
                     return autoFragment(buffer, available);
 
-                // No space in the buffer, so we have to copy the partial payload
+                // No space in the buffer, so we have to copy the partial payload.
+                // The size of this allocation is limited by the maxFrameSize.
                 aggregate = bufferPool.acquire(payloadLength, false);
```

Comment:
Let's use `ByteBufferAccumulator` here rather than an `aggregate` buffer.

Reply:
We are not accumulating ByteBuffers here so this would be pointless. We just want a single buffer to back the websocket Frame.